java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/build/preprocessor/AbstractPreprocessor.java

```diff
@@ -76,11 +224,188 @@ public void run() {
                     """;
         }
-        try {
-            Files.writeString(Paths.get(jbsDirectory.toString(), "Containerfile"), containerFile);
-        } catch (IOException e) {
-            Log.errorf("Unable to write Containerfile", e);
-            throw new RuntimeException(e);
-        }
+
+        return containerFile;
+    }
+
+    /**
+     * This will generate the settings and toolchain into the standard $HOME/.m2 location and configure
+     * altDeploymentDirectory to be used by default.
+     */
+    private String getMavenSetup() {
+        return """
+            echo "MAVEN_HOME:$MAVEN_HOME"
+            PATH="${MAVEN_HOME}/bin:$PATH"
+
+            if [ ! -d "${MAVEN_HOME}" ]; then
+                echo "Maven home directory not found at ${MAVEN_HOME}" >&2
+                exit 1
+            fi
+
+            if [ ! -z ${CACHE_URL+x} ]; then
+            cat >${HOME}/.m2/settings.xml <<EOF
+        <settings>
+          <mirrors>
+            <mirror>
+              <id>mirror.default</id>
+              <url>${CACHE_URL}</url>
+              <mirrorOf>*</mirrorOf>
+            </mirror>
+          </mirrors>
+        EOF
+            else
+                cat >${HOME}/.m2/settings.xml <<EOF
+        <settings>
+        EOF
+            fi
+            cat >>${HOME}/.m2/settings.xml <<EOF
+          <!-- Allows a secondary Maven build to use results of prior (e.g. Gradle) deployment -->
+          <profiles>
+            <profile>
+              <id>alternate</id>
+              <activation>
+                <activeByDefault>true</activeByDefault>
+              </activation>
+              <repositories>
+                <repository>
+                  <id>artifacts</id>
+                  <url>file:///var/workdir/workspace/artifacts</url>
+                  <releases>
+                    <enabled>true</enabled>
+                    <checksumPolicy>ignore</checksumPolicy>
+                  </releases>
+                </repository>
+              </repositories>
+              <pluginRepositories>
+                <pluginRepository>
+                  <id>artifacts</id>
+                  <url>file:///var/workdir/workspace/artifacts</url>
+                  <releases>
+                    <enabled>true</enabled>
+                    <checksumPolicy>ignore</checksumPolicy>
+                  </releases>
+                </pluginRepository>
+              </pluginRepositories>
+            </profile>
+            <profile>
+              <id>deployment</id>
+              <activation>
+                <activeByDefault>true</activeByDefault>
+              </activation>
+              <properties>
+                <altDeploymentRepository>
+                  local::file:///var/workdir/workspace/artifacts
+                </altDeploymentRepository>
+              </properties>
+            </profile>
+          </profiles>
+        </settings>
+        EOF
```

Comment:
The settings.xml for Indy, I took FROM PNC and merged with your
```
<settings>
  <mirrors>
    <mirror>
      <id>indy-mvn</id>
      <mirrorOf>*</mirrorOf>
      <url>
        ${MVN_REPO_DEPENDENCIES_URL}
      </url>
    </mirror>
  </mirrors>

  <servers>
    <server>
      <id>indy-mvn</id>
      <configuration>
        <connectionTimeout>60000</connectionTimeout>
        <httpHeaders>
          <property>
            <name>Authorization</name>
            <value>Bearer ${MVN_TOKEN}</value>
          </property>
        </httpHeaders>
      </configuration>
    </server>
  </servers>

  <proxies>
    <proxy>
      <id>indy-http</id>
      <active>true</active>
      <protocol>http</protocol>
      <host>indy-generic-proxy</host>
      <port>80</port>
      <!-- <username>build-ADDTW3JAGHYAA+tracking</username> -->
      <username>${BUILD_ID}+tracking</username>
      <password>${MVN_TOKEN}</password>
      <nonProxyHosts>indy|localhost</nonProxyHosts>
    </proxy>
    <proxy>
      <id>indy-https</id>
      <active>true</active>
      <protocol>https</protocol>
      <host>indy-generic-proxy</host>
      <port>80</port>
      <username>${BUILD_ID}+tracking</username>
      <password>${MVN_TOKEN}</password>
      <nonProxyHosts>indy|localhost</nonProxyHosts>
    </proxy>
  </proxies>

  <profiles>
    <profile>
      <id>deployment</id>
      <properties>
        <altDeploymentRepository>
          local::file:///var/workdir/workspace/artifacts
        </altDeploymentRepository>
      </properties>
    </profile>
  </profiles>

  <activeProfiles>
    <activeProfile>deployment</activeProfile>
  </activeProfiles>

  <interactiveMode>false</interactiveMode>
  <localRepository>/tmp/.m2/repository</localRepository>

</settings>
```

Reply:
I would bake the ${MVN_REPO_DEPENDENCIES_URL} and ${BUILD_ID} in the OCI source image as that is required for the reproducibility and leave only ${MVN_TOKEN} (probably not the best name) configurable.

Reply:
I think we need to discuss this; I'm not entirely sure I follow or what you're trying to modify?

I don't think the server needs to be configured here as its not deploying? 
I'm not entirely sure about how the proxy section is used either.

Reply:
> I don't think the server needs to be configured here as its not deploying?
I need to verify this, it might be we need auth for read operation from a per-build repo too

> I'm not entirely sure about how the proxy section is used either.
this is a http proxy, used by the tools that are started by maven, eg. a maven task using wget.
Similar to maven there is an http "repo" (persistent cache) per build.

I believe this is all we need for a build using Indy, as I said it's a copy form PNC configured build, except the altDeploymentRepository